src/flamegraph/svg.rs

```diff
@@ -48,224 +48,270 @@ pub(super) struct StyleOptions<'a> {
 
-    // We don't care too much about allocating just for the prelude
-    let mut buf = StrStack::new();
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE + 5,
-            x: (super::IMAGEWIDTH / 2) as f64,
-            y: (super::FONTSIZE * 2) as f64,
-            text: (&*opt.title).into(),
-            location: Some("middle"),
-            extra: None,
-        },
-    )?;
+        svg.write_event(Event::Start(
+            BytesStart::borrowed_name(b"script")
+                .with_attributes(iter::once(("type", "text/ecmascript"))),
+        ))?;
+        svg.write_event(Event::CData(BytesText::from_escaped_str(&format!(
+            "\
+    var nametype = '{}';
+    var fontsize = {};
+    var fontwidth = {};
+    var xpad = {};
+    var inverted = {};
+    var searchcolor = 'rgb(230,0,230)';",
+            self.name_type,
+            self.font_size,
+            self.image_width,
+            super::XPAD,
+            self.direction == Direction::Inverted
+        ))))?;
+        svg.write_event(Event::CData(BytesText::from_escaped_str(include_str!(
+            "flamegraph.js"
+        ))))?;
+        svg.write_event(Event::End(BytesEnd::borrowed(b"script")))?;
 
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE,
-            x: super::XPAD as f64,
-            y: (style_options.imageheight - (super::YPAD2 / 2)) as f64,
-            text: " ".into(),
-            location: None,
-            extra: iter::once(("id", "details")),
-        },
-    )?;
+        svg.write_event(Event::Empty(
+            BytesStart::borrowed_name(b"rect").with_attributes(vec![
+                ("x", "0"),
+                ("y", "0"),
+                ("width", &*format!("{}", self.image_width)),
+                ("height", &*format!("{}", style_options.imageheight)),
+                ("fill", "url(#background)"),
+            ]),
+        ))?;
 
-    write_str(
-        svg,
-        &mut buf,
-        TextItem {
-            color: "rgb(0, 0, 0)",
-            size: super::FONTSIZE,
-            x: super::XPAD as f64,
-            y: (super::FONTSIZE * 2) as f64,
-            text: "Reset Zoom".into(),
-            location: None,
-            extra: vec![
-                ("id", "unzoom"),
-                ("onclick", "unzoom()"),
-                ("style", "opacity:0.0;cursor:pointer"),
-            ],
-        },
-    )?;
+        // We don't care too much about allocating just for the prelude
+        let mut buf = StrStack::new();
+        self.write_str(
+            svg,
+            &mut buf,
+            TextItem {
+                color: "rgb(0, 0, 0)",
+                size: self.font_size + 5,
+                x: (self.image_width / 2) as f64,
+                y: (self.font_size * 2) as f64,
+                text: self.title.clone().into(),
```

Comment:
I don't think the `clone().into()` is needed here? Can it just be `.to_string()` maybe?

Reply:
Me as well, but compiler do not like the proposed either:
```
180 |                 text: &self.title,
    |                       ^^^^^^^^^^^ expected enum `flamegraph::svg::TextArgument`, found reference
    |
    = note: expected type `flamegraph::svg::TextArgument<'_>`
               found type `&std::string::String`
```
I decided to get back to optimisation later.

Reply:
Ah, what's going on here is that you need the `.into()` to get a `TextArgument`, but you _don't_ need the `clone`, since `TextArgument: From<&str>`: https://github.com/jonhoo/inferno/blob/c74e8b9dbd505fb5680212a17025e980ae331edb/src/flamegraph/svg.rs#L17

You may need to make Rust realize that you have a `&str` (not just a `&String`) by doing:
```rust
text: (&*self.title).into()
```

Alternatively you could add an `impl From<&String> for TextArgument` in `svg` :)

Reply:
done